### ISSUE TITLE
giving the inputs it's own thread

### ISSUE THREAD

**Issue by BielBdeLuna** [OpenTechEngine/Discussions#3]
Nvidia details this as an issue to port Doom3BFG to Andorid in this article: https://developer.nvidia.com/content/bringing-doom-3-bfg-edition-nvidia-shield

it seems that the input makes the app to crash when loading a map because the inputs get paused or doesn't keep working and Androd detects it as a crash

maybe we could incorporate Nvidia's solution to the engine?

also in this article it is interesting the reflection on the asset storage: on how it differed in idTechX from idTech4, to store every asset, necessary in a level, packed within the level's pack, therefore there is a lot of repetition of assets. This helped when reading data from an optical disk but in modern flash drives this is no longer necessary.

**Comment by DanielGibson** [comment]
Nope, on some platforms (and with SDL) the input must be polled in the thread that created the window.

Also: Why incorporate a solution for a problem we don't have?

**Comment by BielBdeLuna** [comment]
because we might want to incorporate Android as platform in the future isn't it?

but if you say that SDL already takes charge for this... then it's fine.

**Comment by Yetta1** [comment]
Intel INDE 2015 which has a free professional version available until 31 August includes tools for Android development, it also comes with a Android debugger. INDE works with Visual Studio and Eclipse. It also includes tools for OpenCL. So grab it while it's still free. Full price is $299.

https://software.intel.com/en-us/intel-inde/try-buy
